internal/pkg/reconcilers/bucket/capa/capa_reconciler.go

```diff
@@ -0,0 +1,157 @@
+package capa
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	capa "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.io/giantswarm/object-storage-operator/api/v1alpha1"
+	"github.io/giantswarm/object-storage-operator/internal/pkg/managementcluster"
+	"github.io/giantswarm/object-storage-operator/internal/pkg/services/s3"
+)
+
+// CAPABucketReconciler reconciles Buckets in CAPA environments.
+type CAPABucketReconciler struct {
+	Client            client.Client
+	ManagementCluster managementcluster.ManagementCluster
+}
+
+func (r CAPABucketReconciler) createS3Service(ctx context.Context) (*s3.Service, error) {
+	logger := log.FromContext(ctx)
+
+	awsCluster := &capa.AWSCluster{}
+	err := r.Client.Get(ctx, client.ObjectKey{Name: r.ManagementCluster.Name, Namespace: r.ManagementCluster.Namespace}, awsCluster)
+	if err != nil {
+		logger.Error(err, "Could not find management cluster AWS Cluster CR")
+		return nil, errors.WithStack(err)
+	}
+
+	if awsCluster.Spec.IdentityRef == nil {
+		logger.Info("AWS Cluster has no identityRef set, skipping")
+		return nil, nil
+	}
+
+	managementClusterIdentity := capa.AWSClusterRoleIdentity{}
+	err = r.Client.Get(ctx, client.ObjectKey{Namespace: awsCluster.Namespace, Name: awsCluster.Spec.IdentityRef.Name}, &managementClusterIdentity)
+	if err != nil {
+		logger.Error(err, "Could not find management cluster AWSClusterRoleIdentity CR")
+		return nil, errors.WithStack(err)
+	}
```

Comment:
If would be much better if we could remove the `cluster-api-provider-aws` dependency from this project, [using `unstructured` to fetch these objects](https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/client/example_test.go#L59-L80).

Reply:
Done